web/yo/app/scripts/controllers/gene.js

```diff
@@ -2892,6 +2907,88 @@ angular.module('oncokbApp')
+             *
+             * @param Array list Google drive collaborative list
+             * @param Object unapprovedUuids List of uuids that even unapproved, when calculate the priority should be incldued. This will be used when user approves the section.
+             * @return Object
+             */
+            function getNewPriorities(list, unapprovedUuids) {
+                var priorities = {};
+                var count = 1;
+
+                if (!_.isArray(unapprovedUuids)) {
+                    unapprovedUuids = [];
+                }
+                _.each(list.asArray(), function(treatmentSec, index) {
+                    var name = treatmentSec.name.text;
+                    var uuid = treatmentSec.name_uuid.text;
+                    var isApproved = !treatmentSec.name_review.has('lastReviewed') && !treatmentSec.name_review.get('added');
+                    if (isApproved || unapprovedUuids.indexOf(uuid) !== -1) {
+                        priorities[uuid] = {};
+                        _.each(name.split(','), function(t) {
+                            var treatment = t.trim();
+                            priorities[uuid][treatment] = count;
+                            count++;
+                        });
+                    }
+                });
+                return priorities;
+            }
+
+            /**
+             * Update treatment priority
+             * @param list list Google drive collaborative list
+             * @param integer index Original index
+             * @param integer moveIndex Index is about move before that index
+             * @return Promise
+             */
+            function updatePriority(list, index, moveIndex) {
+                var deferred = $q.defer();
+
+                // if treatment is only moved one position,
+                // only two sections will be affected.
+                // Otherwise, all treatments should be updated.
+                // Update priorities
+                var priorities = getNewPriorities(list);
+                var postData = {};
+
+                index = Number.isInteger(index) ? index : -1;
+                moveIndex = Number.isInteger(moveIndex) ? moveIndex : -1;
+
+                if (Math.abs(index - moveIndex) === 1) {
+                    var indexUUid = list.get(index).name_uuid.text;
+                    var moveIndexUUid = list.get(moveIndex).name_uuid.text;
+
+                    // If one of the section is not approved yet,
+                    // no need to trigger update.
+                    if (priorities[indexUUid] && priorities[moveIndexUUid]) {
+                        postData[indexUUid] = priorities[indexUUid];
+                        postData[moveIndexUUid] = priorities[moveIndexUUid];
+                    }
+                } else {
+                    postData = priorities;
+                }
+
+                if (Object.keys(postData).length > 0) {
```

Comment:
needs to do isDesiredGene and internal network check

Reply:
I can add isDesiredGene, but internal check should be inside databaseconnector as wel discussed before.